samples-generator/bin/3d-tiles-samples-generator.js

```diff
@@ -926,24 +942,60 @@ function saveInstancedTileset(tilesetName, tileOptions, tilesetOptions) {
 
 function saveBatchedTileset(tilesetName, tileOptions, tilesetOptions) {
     var tilesetDirectory = path.join(outputDirectory, 'Batched', tilesetName);
-    var contentUri = lowercase(tilesetName) + '.b3dm';
-    var tilePath = path.join(tilesetDirectory, contentUri);
-    var tilesetPath = path.join(tilesetDirectory, 'tileset.json');
 
     tileOptions = defaultValue(tileOptions, {});
     tileOptions.buildingOptions = defaultValue(tileOptions.buildingOptions, buildingTemplate);
     tileOptions.transform = defaultValue(tileOptions.transform, buildingsTransform);
     tileOptions.relativeToCenter = defaultValue(tileOptions.relativeToCenter, true);
 
     tilesetOptions = defaultValue(tilesetOptions, {});
+
+
+    var ext = '';
+
+    if (argv.glb) {
+        tileOptions.useGlb = true;
+        ext = '.glb';
+    }
+
+    if (argv.gltf) {
+        tileOptions.useGltf = true;
+        ext = '.gltf';
+    }
+
+    if (argv.legacy) {
+        tileOptions.useLegacy = true;
+        ext = '.b3dm';
+    }
+
+    var contentUri = lowercase(tilesetName) + ext;
     tilesetOptions.contentUri = contentUri;
     tilesetOptions.geometricError = smallGeometricError;
     if (!defined(tilesetOptions.region) && !defined(tilesetOptions.sphere) && !defined(tilesetOptions.box)) {
         tilesetOptions.region = smallRegion;
     }
 
+    var rootDirectory = path.join(__dirname, '../');
+    var tilePath = path.join(rootDirectory, tilesetDirectory, contentUri);
+    var tilesetPath = path.join(tilesetDirectory, 'tileset.json');
+
     return createBuildingsTile(tileOptions)
         .then(function(result) {
+            if (argv.glb) {
+                return Promise.all([
+                    fsExtra.outputFile(tilePath, result),
+                    saveJson(tilesetPath, createTilesetJsonSingle(tilesetOptions), prettyJson)
+                ]);
+            }
+
+            else if (argv.gltf) {
+                return Promise.all([
+                    saveJson(tilesetPath, createTilesetJsonSingle(tilesetOptions), prettyJson),
+                    saveJson(tilePath, result, prettyJson)
+                ]);
+            }
+
+            // old .b3dm
             var b3dm = result.b3dm;
             var batchTableJson = result.batchTableJson;
             tilesetOptions.properties = getProperties(batchTableJson);
```

Comment:
The new gltf/glb code paths don't return the batch table json which is needed to generate the `properties` object in `tileset.json`.

Reply:
fixed by moving 
```
var batchTableJson = result.batchTableJson;
var batchTableJson = result.batchTableJson;
tilesetOptions.properties = getProperties(batchTableJson);
```
higher in the function, so that properties is initialized by the time the gltf / glb / tileset with embedded gltf / glb is written.